contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -71,37 +75,27 @@ contract FraudProofFacet is StateChannelCommon {
         return signer1; 
     }
 
-    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
+    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal pure returns (address) {
         BlockEmptyProof memory blockEmptyProof = abi.decode(encodedProof, (BlockEmptyProof));
-        StateSnapshot memory latestStateSnapshot = blockEmptyProof.latestStateSnapshot;
-        bytes32 previousStateSnapshotHash = blockEmptyProof.previousStateSnapshotHash;
-        bytes memory previousStateMachineState = blockEmptyProof.previousStateMachineState;
         Block memory fraudBlock = abi.decode(blockEmptyProof.emptyBlock.encodedBlock, (Block));
+        Block memory previousBlock = abi.decode(blockEmptyProof.previousBlock.encodedBlock, (Block));
```

Comment:
don't decode the previousBlock immediately since there might not be a previous block (if fraudBlock.height == 0), so first check the case blockHeight == 0 and if it's not that than decode previousBlock (previousBlock is optional)